src/mavedb/scripts/map_to_uniprot_id_from_mapped_metadata.py

```diff
@@ -0,0 +1,97 @@
+import click
+import logging
+from typing import Optional
+
+from sqlalchemy.orm import Session
+
+from mavedb.scripts.environment import with_database_session
+from mavedb.models.score_set import ScoreSet
+from mavedb.lib.uniprot.id_mapping import UniProtIDMappingAPI
+from mavedb.lib.uniprot.utils import infer_db_name_from_sequence_accession
+from mavedb.lib.mapping import extract_ids_from_post_mapped_metadata
+
+VALID_UNIPROT_DBS = [
+    "UniProtKB",
+    "UniProtKB_AC-ID",
+    "UniProtKB-Swiss-Prot",
+    "UniParc",
+    "UniRef50",
+    "UniRef90",
+    "UniRef100",
+]
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@with_database_session
+@click.option("--score-set-urn", type=str, default=None, help="Score set URN to process. If not provided, process all.")
+@click.option("--polling-interval", type=int, default=3, help="Polling interval in seconds for checking job status.")
+@click.option("--polling-attempts", type=int, default=5, help="Number of tries to poll for job completion.")
+@click.option("--to-db", type=str, default="UniProtKB", help="Target UniProt database for ID mapping.")
```

Comment:
Maybe worth adding another option to skip target genes that already have a `uniprot_id_from_mapped_metadata` value, or make that the default behavior with an option to override?

Reply:
Yup, that would be useful. I added two new flags:
- `--prefer-swiss-prot` (true by default) prefers swiss prot mapping results when available. If no swiss prot entry is available, it will return all entries. If any are available, it will return them all.
- `--refresh-mapped-identifier` (false by default) will refresh the mapped uniprot id if it exists. If this flag is off and an identifier exists, the target will be skipped.